WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Amount/SimplePaymentsAmountViewModel.swift

```diff
@@ -150,13 +119,23 @@ final class SimplePaymentsAmountViewModel: ObservableObject {
         analytics.track(event: WooAnalyticsEvent.PaymentsFlow.paymentsFlowCanceled(flow: .simplePayment))
     }
 
+
+}
+
+private extension SimplePaymentsAmountViewModel {
     /// Updates the initial order status.
     ///
-    private func updateInitialOrderStatus() {
+    func updateInitialOrderStatus() {
         NewOrderInitialStatusResolver(siteID: siteID, stores: stores).resolve { [weak self] baseStatus in
             self?.initialOrderStatus = baseStatus
         }
     }
+
+    func listenToAmountChanges() {
+        formattableAmountTextFieldViewModel.$amount.map { _ in
+            !self.formattableAmountTextFieldViewModel.amountIsValid
+        }.assign(to: &$shouldDisableDoneButton)
+    }
```

Comment:
Same as above, perhaps we could split the method into `binding/listenToChanges` and `validation`?